streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamFlatTransformTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.kstream.TransformerSupplier;
+import org.apache.kafka.streams.kstream.internals.KStreamFlatTransform.KStreamFlatTransformProcessor;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockSupport;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class KStreamFlatTransformTest extends EasyMockSupport {
+
+    private Number inputKey;
+    private Number inputValue;
+
+    private Transformer<Number, Number, Iterable<KeyValue<Integer, Integer>>> transformer;
+    private ProcessorContext context;
+
+    private KStreamFlatTransformProcessor<Number, Number, Integer, Integer> processor;
+
+    @Before
+    public void setUp() {
+        inputKey = 1;
+        inputValue = 10;
+        Transformer<Number, Number, Iterable<KeyValue<Integer, Integer>>> tempTransformer = mock(Transformer.class);
+        transformer = tempTransformer;
+        context = strictMock(ProcessorContext.class);
+        processor = new KStreamFlatTransformProcessor<Number, Number, Integer, Integer>(transformer);
+    }
+
+    @Test
+    public void shouldInitialiseFlatTransformProcessor() {
+        transformer.init(context);
+        replayAll();
```

Comment:
I am always unsure: don't we need to call "expectLastCall()" or similar?

Reply:
No, it suffices to write it as I did it in this case. You would use `expectLastCall()` to check whether a method is called multiple times and you do not want to repeat the same line amongst other use cases. See [this example](http://easymock.org/user-guide.html#verification-calls) in the EasyMock Documentation.